### ISSUE TITLE
 First two entries of the w are momenta. Change 6 dimensional wave functions to 4 dimensions (possible?)

### ISSUE THREAD

**Issue by roiser** [roiser/madgraph4gpu#18]

**Comment by valassi** [comment]
Note for instance https://github.com/roiser/madgraph4gpu/blob/812f7438462c01e1052c4be48cafcdaef104195c/examples/gpu/eemumu_AV/SubProcesses/P1_Sigma_sm_epem_mupmum/CPPProcess.cc#L326 : in function FFV1_0, the first two dimensions our of 6 are not used at all.

In https://github.com/roiser/madgraph4gpu/blob/812f7438462c01e1052c4be48cafcdaef104195c/examples/gpu/eemumu_AV/SubProcesses/P1_Sigma_sm_epem_mupmum/CPPProcess.cc#L433 instead, in FFV1P0_3 those two dimensions are simply used to be added (internal particle has momenta that are the sums of incomding particles)

**Comment by oliviermattelaer** [comment]
Yes this is an idea that I had in mind, you can in principle replace those entry by a couple of binary number such that the information can be used if/when needed. (if you pass a binary number like 000101 for each Wavefunction, you can recompute that information from the allmomenta (global memory) when needed. -- the 0/1 indicates if that particle momenta need to be summed or not-- 

This is certainly  not useful for e+ e- > mu+ mu- (since we are compute bound) but for the other processes this can make a lot of sense since we are memory bound

**Comment by oliviermattelaer** [comment]
Hi, 

This is now implemented in the branch ep2_w4.
@roiser  could you run the profiler on it?

Would be nice to see if this has any impact.
By just looking at the performance for g g > t t~ g g this does not seem to improve anything at all.
But having the plot and the nsight-compute will allow to give us a better idea of the situation.
